### PR TITLE
create bare bones product card component

### PR DESCRIPTION
**Pull request by spencerlepine**
Wrote tests for the ProductCard component => **[1.4.1] Related Product Cards**

> See comments for literal business requirements.

Screen Shot 2021-09-13 at 10 18 12 AM

### **Features involved:**
1.4.1.1.  Product Information
1.4.1.2.  Product Preview Images

## **Overview:**
Write an element to accept product details and render them in a `react-bootstrap/Card`

Created some tests for the ProductCard component. This will match the demo shown in the image above.
The test will pass down a product object (see `/ProductCard/sampleProduct.js`)
The ProductCart will use the react-bootstrap Card and render the props.

There are more functions to be used in the ProductCard. The original list of products from '/api/products/'
does not contain an object with ALL product details. For example, the thumbnail images are not included. Once the card renders, it will need to fetch more details from the API.

### CODE REVIEW

**Comment by spencerlepine** [comment]
## [1.4.1] Related Product Cards
The related product lists will consist of cards.  Each card will display the information for a single product.  
The card itself will be clickable. Clicking the card will navigate to the detail page for that product.

**Comment by spencerlepine** [comment]
## [1.4.1.1]  Product Information
The following information will appear on the card.  This information will all be read-only and will not have any interactivity associated.
Product Category 
Product Name
Price - As the price is not actually derived from the product, the price displayed should be that for the default style. Sale prices should be reflected.  If the style is currently discounted, then the sale price should appear in red, followed by the original price which is struckthrough.
Star Rating (# of Reviews) - Each product has an average rating based on its reviews.  The average rating of the product will be displayed in the format of solid or outlined stars, where the solid stars represents the review score. A total of 5 stars should always appear, and the amount filled in should correspond to the average score.  
The visual for rating should be representative of up to a quarter of a review point.  For example, if the average is 3.8, this should display as 3¾ solid stars and 1¼ outlined stars.  If there are no reviews, this entire section should be hidden.

**Comment by spencerlepine** [comment]
## [1.4.1.2]  Product Preview Images
The product card should display preview images of the related products.  The images which appear on the product card should be the same that appear in the Overview module on the item detail page for that product. 
By default, the preview image displayed on each card will be the primary image for that product.  This should be the same which first appears on the image detail page’s image gallery.

File: client/src/Components/RelatedContainer/CardList/ProductCard/ProductCard.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
```
Comment (ggbbi) [inline client/src/Components/RelatedContainer/CardList/ProductCard/ProductCard.test.js:2]: remember to change to import from `../test-utils.jsx`

File: client/src/Components/RelatedContainer/CardList/ProductCard/ProductCard.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import ProductCard from './ProductCard.jsx';
+import sampleProduct from './sampleProduct';
```
Comment (ggbbi) [inline client/src/Components/RelatedContainer/CardList/ProductCard/ProductCard.test.js:4]: missing file extension(?)

**Review by ggbbi — APPROVED** [review]

**Pull request merged**